File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorDIAntipattern/_2_ServiceLocator/ApplicationRoot.cs
using System;
using Microsoft.Practices.Unity;
using ServiceLocatorDIAntipattern._2_ServiceLocator.Core;
using ServiceLocatorDIAntipattern._2_ServiceLocator.InMessages;
using ServiceLocatorDIAntipattern._2_ServiceLocator.Inbound;
using ServiceLocatorDIAntipattern._2_ServiceLocator.Outbound;
using ServiceLocatorDIAntipattern._2_ServiceLocator.Services;

namespace ServiceLocatorDIAntipattern._2_ServiceLocator
{
    public class ApplicationRoot
    {
      static ApplicationRoot()
      {
        Context.RegisterType<IProcessingWorkflow, AcmeProcessingWorkflow>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IInbound, BinaryUdpInbound>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IInputSocket, UdpSocket>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IOutputSocket, TcpSocket>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IPacketParsing, BinaryParsing>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IOutbound, Outbound.Outbound>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IAuthorization, ActiveDirectoryBasedAuthorization>(new ContainerControlledLifetimeManager());
        Context.RegisterType<IRepository, MsSqlBasedRepository>(new ContainerControlledLifetimeManager());
        // forgot about this... Context.RegisterType<IMarshalling, XmlMarshalling>(new ContainerControlledLifetimeManager());

        Context.RegisterType<SqlDataDestination>(new ContainerControlledLifetimeManager());
        
        //not container controlled
        Context.RegisterType<IOutboundMessage, OutboundMessage>();
        Context.RegisterType<Random>(new InjectionFactory(container => new Random()));
        Context.RegisterType<NullMessage>();
        Context.RegisterType<StartMessage>();
        Context.RegisterType<StopMessage>();
      }

      public static void Main(string[] args)
      {
        try
        {
          var sys = new TeleComSystem(); //uses container inside, but should be resolved itself!
          sys.Start();

        }
        finally
        {
          Context.Dispose();
        }
      }

      //simplified
      public static final UnityContainer Context = new UnityContainer();
    }




}

File: Java/dependency-injection/_06_conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/ApplicationRoot.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Core.AcmeProcessingWorkflow;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Core.ProcessingWorkflow;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Core.TeleComSystem;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.NullMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.StartMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.StopMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Inbound.*;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Outbound.*;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Services.*;

import java.util.Random;

public class ApplicationRoot {
  public static final ConformingContainer CONTEXT = createContainer();

  private static ConformingContainer createContainer() {
    CONTEXT.as(ProcessingWorkflow.class).useSingle(AcmeProcessingWorkflow.class);
    CONTEXT.as(Inbound.class).useSingle(BinaryUdpInbound.class);
    CONTEXT.as(InputSocket.class).useSingle(UdpSocket.class);
    CONTEXT.as(OutputSocket.class).useSingle(TcpSocket.class);
    CONTEXT.as(PacketParsing.class).useSingle(BinaryParsing.class);
    CONTEXT.as(Outbound.class).useSingle(MessageOutbound.class);
    CONTEXT.as(Authorization.class).useSingle(ActiveDirectoryBasedAuthorization.class);
    CONTEXT.as(Repository.class).useSingle(MsSqlBasedRepository.class);
    CONTEXT.as(Marshalling.class).useSingle(XmlMarshalling.class);

    CONTEXT.useSingle(SqlDataDestination.class);

    //not container controlled
    CONTEXT.as(OutboundMessage.class).useCreated(XmlOutboundMessage.class);
    CONTEXT.as(Random.class).useCreated(Random.class);
    CONTEXT.useCreated(NullMessage.class);
    CONTEXT.useCreated(StartMessage.class);
    CONTEXT.useCreated(StopMessage.class);
    return CONTEXT;
  }

  public static void Main(String[] args) throws Exception {
    try {
      TeleComSystem sys = new TeleComSystem(); //uses container inside, but should be resolved itself!
      sys.start();
    } finally {
      CONTEXT.close();
    }
  }
}


